### PR TITLE
[iOS] Free up resources upon map widget disposal

### PR DESCRIPTION
**Pull request by evil159**
### What does this pull request do?

Adds tear down for Pigeon-generated channel handlers to reduce memory consumption when the platform FlutterView goes out of scope.

Here's chart of memory consumption opening/closing a screen with map view couple of times:

  | Before | After |
  | ----- | ----- |
  |  |  |

Before the resources from previous map widget were lingering until a new map widget was about to be created(you can see momentary dips in the "Before" chart just before map screen).

### What is the motivation and context behind this change?

https://github.com/mapbox/mapbox-maps-flutter/issues/380

### Pull request checklist:
 - [x] Add a changelog entry.
 - [ ] Not testable.

### CODE REVIEW

**Review by maios — APPROVED** [review]

**Comment by sepios87** [comment]
The code in the ios/Classes/MapboxMapController.swift file is commented out even though it allows freeing up memory. I tested this implementation on 0.5.1 by adapting the names and this implementation does correctly fix crashes related to memory overuse on iOS. I might have changed the name of releaseMethodChannels to something more descriptive but personally I would approve this pull request !

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapController.kt [hunk 1]
```diff
@@ -181,6 +183,10 @@ class MapboxMapController(
         gestureController.removeListeners()
         result.success(null)
       }
+      "platform#releaseMethodChannels" -> {
+        releaseMethodChannels()
```
Comment (sepios87) [inline android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapController.kt:187]: It would just be necessary to change the name releaseMethodChannels to a more explicit name

File: example/integration_test/style/style_test.dart [hunk 2]
```diff
@@ -283,15 +283,25 @@ void main() {
 
     expect((await style.getStyleLightProperty("flat-light-id", "color")).value,
         listCloseTo(Colors.red.toRGBAList(), 0.0001));
-    expect((await style.getStyleLightProperty("flat-light-id", "color-transition")).value,
+    expect(
```
Comment (sepios87) [inline example/integration_test/style/style_test.dart:286]: revert to the old formatting, there is no point in changing this file

File: ios/Classes/MapboxMapController.swift [hunk 3]
```diff
@@ -186,11 +185,30 @@ class MapboxMapController: NSObject, FlutterPlatformView {
...
+        case "platform#releaseMethodChannels":
+            releaseMethodChannels()
+            result(nil)
         default:
             result(FlutterMethodNotImplemented)
         }
     }
 
+    private func releaseMethodChannels() {
+//        channel.setMethodCallHandler(nil)
```
Comment (sepios87) [inline ios/Classes/MapboxMapController.swift:197]: uncomment this

File: lib/src/location_settings.dart [hunk 4]
```diff
@@ -19,16 +19,22 @@ class LocationSettings {
   Future<void> updateSettings(LocationComponentSettings settings) async {
     if (settings.locationPuck == null) {
       // If locationPuck is not set, fallback to use DefaultLocationPuck2D.
-      settings.locationPuck = LocationPuck(locationPuck2D: DefaultLocationPuck2D());
+      settings.locationPuck =
```
Comment (sepios87) [inline lib/src/location_settings.dart:22]: revert to the old formatting, there is no point in changing this file

File: lib/src/mapbox_map.dart [hunk 5]
```diff
@@ -144,8 +144,8 @@ class MapboxMap extends ChangeNotifier {
       StyleManager(binaryMessenger: _proxyBinaryMessenger);
 
   /// The interface to set the location puck.
-  late LocationSettings location =
-      LocationSettings(_LocationComponentSettingsInterface(
+  late LocationSettings location = LocationSettings(
```
Comment (sepios87) [inline lib/src/mapbox_map.dart:147]: revert to the old formatting, there is no point in changing this part

File: lib/src/mapbox_maps_platform.dart [hunk 6]
```diff
@@ -137,20 +137,12 @@ class _MapboxMapsPlatform {
...
   }
 
-  void dispose() {
+  void dispose() async {
+    await _channel.invokeMethod('platform#releaseMethodChannels');
     _suffixesRegistry.releaseSuffix(_channelSuffix);
     _channel.setMethodCallHandler(null);
   }
 
-  Future<void> addEventListener(_MapEvent event) async {
```
Comment (sepios87) [inline lib/src/mapbox_maps_platform.dart:145]: I do not know if it is useful

File: lib/src/pigeons/gesture_listeners.dart [hunk 7]
```diff
@@ -10,6 +10,7 @@ List<Object?> wrapResponse(
   }
   return <Object?>[error.code, error.message, error.details];
 }
+
```
Comment (sepios87) [inline lib/src/pigeons/gesture_listeners.dart:13]: revert this change

**Review by sepios87 — COMMENTED** [review]
Please review the comments but otherwise good job fixing memory leaks on iOS I confirm!
